File: wiki-java/src/main/java/dev/freya02/wiki/filters/GeneralChannelFilter.java
package dev.freya02.wiki.filters;

import io.github.freya022.botcommands.api.components.ComponentInteractionFilter;
import io.github.freya022.botcommands.api.core.service.annotations.BService;
import net.dv8tion.jda.api.events.interaction.component.GenericComponentInteractionCreateEvent;
import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

// --8<-- [start:component_filter-java]
@BService
@NullMarked // Everything is non-null unless @Nullable
public class GeneralChannelFilter implements ComponentInteractionFilter {
    private static final long CHANNEL_ID = 722891685755093076L;

    @Override
    public boolean getGlobal() {
        // So we can apply this filter on specific components
        return false;
    }

    @Nullable
    @Override
    public String check(GenericComponentInteractionCreateEvent event, @Nullable String handlerName) {
        if (event.getChannelIdLong() != CHANNEL_ID) {
            event.reply("This button can only be used in <#" + CHANNEL_ID + ">")
                    .setEphemeral(true)
                    .queue();
            return "Button was used in the wrong channel";
        }

        // Correct channel, return no error message
        return null;
    }
}
// --8<-- [end:component_filter-java]

File: wiki-kotlin/src/main/kotlin/dev/freya02/wiki/filters/GeneralChannelFilter.kt
package dev.freya02.wiki.filters

import dev.freya02.botcommands.jda.ktx.coroutines.await
import dev.freya02.botcommands.jda.ktx.messages.reply_
import io.github.freya022.botcommands.api.components.ComponentInteractionFilter
import io.github.freya022.botcommands.api.core.service.annotations.BService
import net.dv8tion.jda.api.events.interaction.component.GenericComponentInteractionCreateEvent

// --8<-- [start:component_filter-kotlin]
@BService
class GeneralChannelFilter : ComponentInteractionFilter {
    private val channelId = 722891685755093076

    // So we can apply this filter on specific components
    override val global: Boolean = false

    override suspend fun checkSuspend(
        event: GenericComponentInteractionCreateEvent,
        handlerName: String?
    ): String? {
        if (event.channelIdLong == channelId) {
            event.reply_("This button can only be used in <#$channelId>", ephemeral = true).await()
            return "Button was used in the wrong channel"
        }
        return null
    }
}
// --8<-- [end:component_filter-kotlin]
